### PR TITLE
Working branch

### PR DESCRIPTION
**Pull request by thomaswatsonv1**
### JIRA link (if applicable) ###
N/A

### Change description ###
Add the replace token step back into the pipeline and increase the vm size and spec. This is the currently deployed version.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X ] No
```

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -27,4 +27,5 @@ override.tf.json
 # !example_override.tf
 
 # Include tfplan files to ignore the plan output of command: terraform plan -out=tfplan
-# example: *tfplan*
\ No newline at end of file
+# example: *tfplan*
+terraform/set_up copy.sh
```
Comment (chris-pateman) [inline .gitignore:31]: I would not have a space in a file name

File: azure-pipelines.yaml [hunk 2]
```diff
@@ -83,37 +84,41 @@ stages:
...
-              parameters:
-                terraformVersion: $(tfVersion)
-                subscription: $(subscription)
-                resourceGroup: $(resourceGroup)
-                location: $(location)
-                storageAccount: $(storageAccount)
-                containerName: $(containerName)
-                stateKey: $(stateKey)-AzureDevOpsAgent.tfstate
-                workingDirectory: $(workDir)
+          - template: ado-pipeline/jobs/replace-token.yaml
```
Comment (chris-pateman) [inline azure-pipelines.yaml:98]: do we have this extension added in and would this be better?
https://marketplace.visualstudio.com/items?itemName=qetza.replacetokens

File: terraform/02-kv.tf [hunk 3]
```diff
@@ -40,7 +40,11 @@ resource "azurerm_key_vault" "keyvault_ado_agent" {
...
   name            = "vh-devops-agent-password"
   value           = random_password.password.result
-  content_type    = "password"
+  #content_type    = "password"
   key_vault_id    = azurerm_key_vault.keyvault_ado_agent.id
-  expiration_date = "2023-05-31T00:00:00Z" # PASSWORD WILL EXPIRE 31st May 2023 ###
+  #expiration_date = "2023-05-31T00:00:00Z" # PASSWORD WILL EXPIRE 31st May 2023 ##
+
+  lifecycle {
+    ignore_changes = [value]
```
Comment (chris-pateman) [inline terraform/02-kv.tf:48]: why would we ignore the value?
if terraform changes this then we will not have the latest version

**Review by chris-pateman — CHANGES_REQUESTED** [review]

**Comment by thomaswatsonv1** [comment]
Hi Chris.

I'm just trying to make the current master a working version so we can start on the SDS version by branching a 'working' master, lol!

The .gitignore was a remnant of a file that has long since been deleted! Deleted the reference in the .gitignore.

The replace token stage will be doing two replaces in future - it will replace the parameterised devops url as well as the pat token. I'm not sure that extension would be more useful or more lightweight than using the bash, but I can look at it for the SDS infra. It seems very json/xml/yaml focussed but that shouldn't be an issue. It will have to run the stage/job twice to do multiple replaces, however.

The lifecycle is there because if, say, the setup shell script failed during a run, then a new password will be created when running the pipeline to fix the shell script. As we've only given read/write permissions, the pipeline will fail because the key vault will already have a value for "vh-ado-devops-password" and it can't delete or purge the original one to add in the new one. The lifecycle just stops a re-creation of the password every run but will allow a new password on a fresh deployment. Does that make sense? Stops a lot of ball-ache during writing/testing.

Hope that makes sense?

Tom

**Pull request closed without merging**